Atos20/indecision-maker-server
# Express ng

### PR Template:

#### What's this PR do?  

+ Adds route handlers to retrieve all music, podcasts, movies, board games, and card games
+ Adds route handlers to retrieve music, podcasts, and movies by genre
+ Removes unnecessary students data from starter repo
+ Removes timestamps from all activity schemas (migration files)
+ Adds genre column to movies table
+ Updates test suite in `app.test.js` with changes needed to search by the queried genre, especially when the genre value has more than one word in its string. 

#### Where should the reviewer start?  

Start in `app.js` and `app.test.js` to see all route handlers and their corresponding tests.

#### How should this be manually tested?  

This can be pulled down locally, run `knex migrate:latest` and `knex seed:run` to get the data up and running in the database via Postico.  Run `npm test` in the terminal to see 11 passing tests.  These can be manipulated to ensure there aren't any false positives.

#### Any background context you want to provide?  

We removed the time stamps from all of the schemas because the `created_at` and `updated_at` properties were not showing as having the correct data type.  The test was receiving a string, while the expected value was a date object.  We tried to manipulate the data so that the expected value would be a string, but our efforts did not lead to a feasible solution.  Instead, we removed the time stamps from all of the tables because we realized we don't have a clear use for them.  If, in the future, we decide it would be worthwhile to tackle this error, we plan to try to change the data type of the time stamps before it gets inserted into the tables.  

We also want to add more sad path testing and wonder if there is a way to test the `catch` in the event of a `500 response status`.

#### What are the relevant tickets?  

Closes #20 

#### Screenshots (if appropriate) 
![Screen Shot 2020-10-29 at 2 55 52 PM](https://user-images.githubusercontent.com/62262404/97631618-de783080-19f6-11eb-97d6-4f3a00f52aa5.png)
